### PR TITLE
chore: Test HybridSDK subspec 

### LINKED ISSUE #3021: Test Cocoapod with HybridSDK target
### Description

We need to create a CI job to test cocoapod `Sentry/HybridSDK` target.

This was raised at https://github.com/getsentry/sentry-cocoa/pull/3017

### PR DESCRIPTION
**Pull request by brustolin**
Added a workflow to test whether `Sentry/HybridSDK` podspec dependency is compiling properly for hybrid SDKs.

This was raised because of #3017

closes #3021 

_#skip-changelog_

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -57,6 +57,16 @@ jobs:
       - name: Validate Podspec
         run: ./scripts/pod-lib-lint.sh ${{ matrix.platform }} ${{ matrix.podspec}} ${{ matrix.library_type}}
 
+  lint-hybridsdk-podspec:
+    name: pod lint Sentry/HybridSDK
+    runs-on: macos-12
```
Comment (philipphofmann) [inline .github/workflows/lint.yml:62]: `m`: macos-13 is way faster.
```suggestion
    runs-on: macos-13
```

**Review by philipphofmann — APPROVED** [review]
Before merging this, please add the error fixed in https://github.com/getsentry/sentry-cocoa/pull/3017 back, push the branch, and check if the newly added CI job fails. Then we can merge this. Thanks 👍

**Comment by brustolin** [comment]
> Before merging this, please add the error fixed in #3017 back, push the branch, and check if the newly added CI job fails. Then we can merge this. Thanks

[Fail confirm](https://github.com/getsentry/sentry-cocoa/actions/runs/4996313772/jobs/8949338345?pr=3031)

**Review by philipphofmann — APPROVED** [review]
Thanks @brustolin 👍

**Pull request merged**